vm-setup/roles/v1aX_integration_test/tasks/provision_cluster.yml

```diff
@@ -1,12 +1,21 @@
 ---
-  - name: Install openshift client
+  - name: Install openshift client in CentOS 8 using pip3
     pip:
       executable: "pip3"
       name: openshift
       state: present
     become: true
     become_user: root
-    when: ansible_facts['distribution'] == "CentOS"
+    when: ansible_facts['distribution'] == "CentOS" and ansible_facts['distribution_major_version'] == '8'
+
+  - name: Install openshift client in CentOS 7 using pip
+    pip:
+      executable: "pip"
```

Comment:
Could you please share why is it required to use pip instead of pip3?

Reply:
Yes, this issue happens when the physical server which is running libvirtd is CentOS7. In this case, Ansible is running python2, therefore if you use pip3 to install the openshift client (provision_cluster playbook) since the ansible_python_interpreter is set to python2/pip2 it won't find the kubernetes module:

```sh
TASK [v1aX_integration_test : Provision cluster] **************************************************************************************************************
task path: /home/alosadag/metal3-dev-env/vm-setup/roles/v1aX_integration_test/tasks/provision_cluster.yml:17
<localhost> ESTABLISH LOCAL CONNECTION FOR USER: alosadag
<localhost> EXEC /bin/sh -c 'echo ~alosadag && sleep 0'
<localhost> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /home/alosadag/.ansible/tmp/ansible-tmp-1584957211.06-58949636475436 `" && echo ansible-tmp-1584957211.06-58949636475436="` echo /home/alosadag/.ansible/tmp/ansible-tmp-1584957211.06-58949636475436 `" ) && sleep 0'
Using module file /usr/lib/python2.7/site-packages/ansible/modules/clustering/k8s/k8s.py
<localhost> PUT /home/alosadag/.ansible/tmp/ansible-local-45010DrKySv/tmpL17g6P TO /home/alosadag/.ansible/tmp/ansible-tmp-1584957211.06-58949636475436/AnsiballZ_k8s.py
<localhost> EXEC /bin/sh -c 'chmod u+x /home/alosadag/.ansible/tmp/ansible-tmp-1584957211.06-58949636475436/ /home/alosadag/.ansible/tmp/ansible-tmp-1584957211.06-58949636475436/AnsiballZ_k8s.py && sleep 0'
<localhost> EXEC /bin/sh -c '/usr/bin/python /home/alosadag/.ansible/tmp/ansible-tmp-1584957211.06-58949636475436/AnsiballZ_k8s.py && sleep 0'
<localhost> EXEC /bin/sh -c 'rm -f -r /home/alosadag/.ansible/tmp/ansible-tmp-1584957211.06-58949636475436/ > /dev/null 2>&1 && sleep 0'
The full traceback is:
Traceback (most recent call last):
  File "/tmp/ansible_k8s_payload_tnA_vG/ansible_k8s_payload.zip/ansible/module_utils/k8s/common.py", line 33, in <module>
    import kubernetes
ImportError: No module named kubernetes
fatal: [localhost]: FAILED! => {
    "changed": false, 
    "error": "No module named kubernetes", 
    "invocation": {
        "module_args": {
            "api_key": null, 
            "api_version": "v1", 
            "append_hash": false, 
            "apply": false, 
            "ca_cert": null, 
            "client_cert": null, 
            "client_key": null, 
            "context": null, 
            "force": false, 
            "host": null, 
            "kind": null, 
            "kubeconfig": null, 
            "merge_type": null, 
            "name": null, 
            "namespace": "metal3", 
            "password": null, 
            "proxy": null, 
            "resource_definition": null, 
            "src": "/tmp/cluster.yaml", 
            "state": "present", 
            "username": null, 
            "validate": null, 
            "validate_certs": null, 
            "wait": false, 
            "wait_condition": null, 
            "wait_sleep": 5, 
            "wait_timeout": 120
        }
    }, 
    "msg": "Failed to import the required Python library (openshift) on eko1.cloud.lab.eng.bos.redhat.com's Python /usr/bin/python. Please read module documentation and install in the appropriate location. If the required library is installed, but Ansible is using the wrong Python interpreter, please consult the documentation on ansible_python_interpreter"
}
```

I found two ways to fix it. One is the selecting the executable based on the major OS version. The other is setting the ansible_python_interpreter to python3 in the inventory:

```
$ cat  vm-setup/inventory.ini 

[virthost]
localhost ansible_python_interpreter=/usr/bin/python3
```

Which one do you think is better fix? @fmuyassarov @maelk

Reply:
Note as well, that CentOS 7 should be decomissioned in favour of CentOS 8, right?

Reply:
From my point of view, I think it is better to use: `ansible_python_interpreter=/usr/bin/python3`, because if you add this it is anyway generic and will not hurt other OS based metal3-dev-env (where the libvirt is running) cases such as Ubuntu/Centos8 based.

Reply:
Ok dropped the commit that includes the conditional statement and added the ansible_python_interpreter to python3 in the inventory

Reply:
mmmm found the following error when running make. Probably it is safer to include the conditional instead of changing the interpreter. Otherwise we will need to replace yum by dnf and probably more things. 

```
 TASK [packages_installation : Install packages on rhel7] ******************************************************************************************************
fatal: [localhost]: FAILED! => {
    "changed": false, 
    "invocation": {
        "module_args": {
            "allow_downgrade": false, 
            "autoremove": false, 
            "bugfix": false, 
            "conf_file": null, 
            "disable_excludes": null, 
            "disable_gpg_check": false, 
            "disable_plugin": [], 
            "disablerepo": [], 
            "download_dir": null, 
            "download_only": false, 
            "enable_plugin": [], 
            "enablerepo": [], 
            "exclude": [], 
            "install_repoquery": true, 
            "install_weak_deps": true, 
            "installroot": "/", 
            "list": null, 
            "lock_timeout": 30, 
            "name": [
                "curl", 
                "dnsmasq", 
                "NetworkManager", 
                "nmap", 
                "OVMF", 
                "patch", 
                "psmisc", 
                "vim-enhanced", 
                "wget", 
                "ansible", 
                "bind-utils", 
                "jq", 
                "libguestfs-tools", 
                "redhat-lsb-core", 
                "unzip", 
                "genisoimage", 
                "qemu-kvm", 
                "libvirt", 
                "libvirt-devel", 
                "libvirt-daemon-kvm", 
                "libvirt-python", 
                "libguestfs-tools", 
                "python-lxml", 
                "polkit-pkla-compat", 
                "python-netaddr", 
                "virt-install", 
                "python-six"
            ], 
            "releasever": null, 
            "security": false, 
            "skip_broken": false, 
            "state": "present", 
            "update_cache": false, 
            "update_only": false, 
            "use_backend": "auto", 
            "validate_certs": true
        }
    }, 
    "msg": "The Python 2 bindings for rpm are needed for this module. If you require Python 3 support use the `dnf` Ansible module instead.. The Python 2 yum module is needed for this module. If you require Python 3 support use the `dnf` Ansible module instead."
}

```

Reply:
Yes now I see. I had the same, one solution would be first to set alternatives to python3 and then modify inventory accordingly. 
But perhaps better to use your initial commit, which adds  conditional statement, otherwise there will be a need for more changes.
Sorry, for making extra work.